docs/CONTRIBUTING.md

```diff
@@ -38,12 +48,79 @@ template will be requested to be updated. Any pull requests opened without a
 corresponding issue may be delayed or be required to undergo further changes
 before being accepted.
 
-### Issues
+#### Pull Request Workflow
 
-All issues should follow our [ISSUE_TEMPLATE](ISSUE_TEMPLATE.md). It is
-additionaly recommended to prefix your issue with an appriopriate "tag" such as
-`[QUESTION]`, `[BUG]`, `[REGRESSION]`, `[IDEA]`, or `[PROPOSAL]`.
+As workflow for Pull Requests (PR) it is recommended to 
 
-### Questions
+1. Fork the [repository](https://github.com/terrafx/terrafx.interop.windows) 
+Make sure the git configuration has the upstream master set, 
+e.g. run `git remote add upstream https://github.com/terrafx/terrafx.interop.windows`
 
-Feel free to open an issue prefixed with `[QUESTION]`.
+2. Create a branch with name reflecting the issue/proposal you are planning to work on.
+
+3. Implement your PR
+
+4. git Commit, Fetch --all, Rebase, Push, and then submit your PR. 
+The submit can be done via git commandline, or from your git gui tool 
+or with the button on the github page of your fork. 
+
+5. If on your own or prompted by a PR review you make further changes, 
+then you can update your PR via Commit, Fetch --all, Rebase, Push. 
+The push will be automatically recognized by github and github 
+will update the PR on the main upstream repository.
+
+6. While one PR is in the works like this and maybe you are waiting on 
+a review you can make another branch for another PR and work on that in parallel.
+
+7. When a PR is approved and closed your code becomes part of the upstream code base 
+and there is no need to keep the branch. 
+Feel free to delete the branch to reduce clutter in your git log, 
+but this step is not required.
+
+#### Coding Conventions
+
+TerraFX follows the industry standard 
+[C# coding conventions as published by Microsoft]
+(https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions) 
+and as enforced by the messages in Visual Studio 2019 
+when configured as *build + intellisense* .  
+Commits should be done such that under these settings there are no warnings.
+
+In addition, there are a few more conventions:
+
+1. **Property vs Method:** a property is auto evaluated by Visual Studio 
```

Comment:
While this is generally good advice, it doesn't belong here. `TerraFX.Interop.*` are all generally interop bindings and so they break almost all of the standard coding conventions/patterns.

Instead, rules like this should belong in a more general doc describing how devs submitting API proposals should consider the shape of the API before logging an issue.

Reply:
Well, they are things that I try to go by as much as possible. Applies more to the samples than to the core of TerraFX. So I'll remove it. Was there in part to get your feedback on it. So it will be part of my PRs, but not of this doc.